kivy/uix/behaviors/knspace.py

```diff
@@ -0,0 +1,433 @@
+            if knspace:
+                value = knspace.fork()
+            else:
+                raise ValueError('Cannot fork with no namesapce')
+
+        for obj, prop_name, uid in self.__callbacks or []:
+            obj.unbind_uid(prop_name, uid)
+        self.__last_knspace = self.__callbacks = None
+
+        if name:
+            if value is None:  # if None, first update the recursive knspace
+                knspace = self.__set_parent_knspace()
+                if knspace:
+                    setattr(knspace, name, self)
+                self._knspace = None  # cause a kv trigger
+            else:
+                setattr(value, name, self)
+                knspace = self._knspace = value
+
+            if not knspace:
+                raise ValueError('Object has name "{}", but no namespace'.
+                                 format(name))
+        else:
+            if value is None:
+                self.__set_parent_knspace()  # update before trigger below
+            self._knspace = value
+
+    knspace = AliasProperty(
+        _get_knspace, _set_knspace, bind=('_knspace', ), cache=False,
+        rebind=True, allownone=True)
+    '''The namespace instance, :class:`KNSpace`, associated with this widget.
+    When this widget is named with :attr:`name` the name is added to the
+    :attr:`knspace` namespace pointing to this widget.
+
+    If the namespace has been set with a :class:`KNSpace` instance, e.g. with
+    `self.knspace = ...`, then that instance is used. Otherwise, we look at
+    the property named :attr:`knspace_key` of this obj. If that object has a
+    knspace property we use that namespace. Otherwise, we look at its
+    :attr:`knspace_key` object and walk up the parent tree until we find
+    a parent who has a namespace instance. Finally, if there's no parent with
+    a namespace, the default :attr:`~cutils.knspace.knspace` namespace is used.
+
+    Both `rebind` and `allownone` are `True`.
+    '''
+
+    knspace_key = StringProperty('parent', allownone=True)
+    '''The name of the property of this instance, to use to find the namespace
+    associated with this instance. Defaults to `'parent'` so that we'll look
+    up the parent tree to find the namespace. See :attr:`knspace`.
+
+    When `None`, we won't search the parent tree for the namespace.
+    `allownone` is `True`.
+    '''
+
+    def _get_name(self):
+        return self._name
+
+    def _set_name(self, value):
+        old_name = self._name
+        knspace = self.knspace
+        if old_name and knspace and getattr(knspace, old_name) == self:
+            setattr(knspace, old_name, None)
+
+        self._name = value
+        if value:
+            if knspace:
+                setattr(knspace, value, self)
+            else:
+                raise ValueError('Object has name "{}", but no namespace'.
+                                 format(value))
+
+    name = AliasProperty(_get_name, _set_name, bind=('_name', ), cache=False)
```

Comment:
Wouldn't this conflict with `Screen.name` when used with Screen? How about `knsname` or `kns_name`?